airflow/providers/common/sql/operators/sql.py

```diff
@@ -60,6 +74,12 @@ def _get_failed_checks(checks, col=None):
     ]
 
 
+def _raise_exception(exception_string, retry_on_failure):
+    if retry_on_failure:
+        raise AirflowException(exception_string)
```

Comment:
This method raises the same exception, with or without retry_on_failure being True

Reply:
This method comes from this [issue](https://github.com/apache/airflow/discussions/26712#discussioncomment-3747662); the `AirflowFailException` should halt the task from retrying.